components/apimgt/org.wso2.carbon.apimgt.core/src/main/java/org/wso2/carbon/apimgt/core/exception/ExceptionCodes.java

```diff
@@ -26,12 +26,13 @@
  */
 public enum ExceptionCodes implements ErrorHandler {
 
-    API_ALREADY_EXISTS(900300, "The API already exists.", 400, " This is error description"),
-    APPLICATION_ALREADY_EXISTS(900301, "The application already exists.", 400, " This is error description"),
-    APIMGT_DAO_EXCEPTION(900302, "Internal server error.", 500, " This is error description"),
-    APIMGT_LIFECYCLE_EXCEPTION(900303, "Life cycle exception occurred", 500, " This is error description"),
-    TIER_CANNOT_BE_NULL(900304, "The tier cannot be null.", 400, " This is error description"),
-    TIER_NAME_INVALID(900305, "The tier name is invalid.", 400, " This is error description"),
+    API_ALREADY_EXISTS(900300, "The API already exists.", 400, " The API already exists"),
+    APPLICATION_ALREADY_EXISTS(900301, "The application already exists.", 400, " The application already exists"),
+    APIMGT_DAO_EXCEPTION(900302, "Internal server error.", 500, " Error occurred while persisting data"),
+    APIMGT_LIFECYCLE_EXCEPTION(900303, "Lifecycle exception occurred", 500, " Error occurred while changing " +
+            "lifecycle state"),
+    TIER_CANNOT_BE_NULL(900304, "The tier cannot be null.", 400, " The tier cannot be null"),
+    TIER_NAME_INVALID(900305, "The tier name is invalid.", 400, " The tier name is invalid"),
     SWAGGER_PARSE_EXCEPTION(900306, "Error while parsing swagger json", 500, "Error while parsing swagger json"),
```

Comment:
I feel this should be 400. Parsing error is due to client is sending an invalid swagger right?